pkg/commands/loaders/branches.go

```diff
@@ -106,6 +106,49 @@ outer:
 	return branches, nil
 }
 
+// Obtain branch information from parsed line output of getRawBranches()
+// split contains the '|' separated tokens in the line of output
+func obtainBranch(split []string) *models.Branch {
+	name := strings.TrimPrefix(split[1], "heads/")
+	branch := &models.Branch{
+		Name:      name,
+		Pullables: "?",
+		Pushables: "?",
+		Head:      split[0] == "*",
+	}
+
+	upstreamName := split[2]
+	if upstreamName == "" {
+		// if we're here then it means we do not have a local version of the remote.
+		// The branch might still be tracking a remote though, we just don't know
+		// how many commits ahead/behind it is
+		return branch
+	}
+
+	track := split[3]
+	re := regexp.MustCompile(`ahead (\d+)`)
+	match := re.FindStringSubmatch(track)
+	if len(match) > 1 {
+		branch.Pushables = match[1]
+	} else {
+		branch.Pushables = "0"
+	}
+
+	re = regexp.MustCompile(`behind (\d+)`)
+	match = re.FindStringSubmatch(track)
+	if len(match) > 1 {
+		branch.Pullables = match[1]
+	} else {
+		branch.Pullables = "0"
+	}
+
+	if track == "[gone]" {
```

Comment:
I know this is gonna increase nesting which some people don't like but we should make it clear that lines 137-143 are mutually exclusive to this code here. I.e. I would first check if the value of `track` is `[gone]` and if not (i.e. in an else block) we check for the ahead/behind counts.

FWIW some other options that came to mind were:
* extract the track-specific code into its own function. I'm not a huge fan of this just because I don't like mutating arguments to a function
* early return to avoid nesting: also not a fan because we'll likely add more stuff to this function over time.

Reply:
I think the nesting here is necessary, so your suggestion is implemented.

Reply:
Could we also move lines 128-135 into the else block as well? I assume we wouldn't have an 'ahead' count if the branch is gone

Reply:
Looks like this comment was missed

Reply:
@john-nguyen09 opened this PR https://github.com/john-nguyen09/lazygit/pull/4 to address this comment.